src/main/java/com/javapractice/interactors/addTodoInteractor.java

```diff
@@ -0,0 +1,26 @@
+package com.javapractice.interactors;
+
+import org.springframework.stereotype.Service;
+
+import com.javapractice.domain.Request.TodoRequest;
+import com.javapractice.domain.Response.TodoResponse;
+import com.javapractice.gateways.ITodoRepository;
+import com.javapractice.interactors.interfaces.IaddTodo;
+
+@Service
+public class addTodoInteractor implements IaddTodo{
+    private final ITodoRepository todoRepository;
+    
+    public addTodoInteractor(ITodoRepository todoRepository) {
+        this.todoRepository = todoRepository;
+    }
+
+    @Override
+    public TodoResponse addTodo(TodoRequest request) {
```

Comment:
This is where you would want to put in guards, but Java does this for you. Basically, your changing from data transfer objects (DTOs) to data access objects (DAOs).